### ISSUE TITLE
Achievement support for epic and gog supported games and download local installers and goodies in the gog library

### ISSUE THREAD

**Issue by MrArdillo** [Heroic-Games-Launcher/HeroicGamesLauncher#1279]

**Comment by MrArdillo** [comment]
Okay, and for the achievements part? Its in process or something?

**Comment by MrArdillo** [comment]
Cool, but in Linux it will work too or is it exclusive for windows because of the service?

**Comment by flavioislima** [comment]
There is no cloud save for Linux so probably only windows games 😕

**Comment by imLinguin** [comment]
well it may be possible, since cloud store is the same for all platforms. We would need to have own API for Linux with locations of saves. Games like Stardew Valley work fine with saves copied from Windows version 😉. We need more research on that part

**Comment by MrArdillo** [comment]
That will be very nice but i refer more of the achievement pop up and in heroic menu like GOG 2.0 in windows in games like the witcher 3 for example

**Comment by yessenbayev** [comment]
Hi everyone,

Sorry for messaging in a dead thread, but I could not find any better information online.

If I launch a game using Heroic, for example on Steam Deck, would Epic Games or GOG correctly register achievements? I don't mean UI responding in any way, I mean that the achievement will be later displayed in Epic Games/GOG if I log in later from a browser, for example.

Thank you so much!

**Comment by imLinguin** [comment]
Epic should register them, as far as I know. GOG needs more research regarding Galaxy Communication Service SDK. Like I said earlier.

**Comment by szszoke** [comment]
> On the GoG Side yes.
> There is work on reimplementing the necessary service, called GalaxyCommunications.exe in the official Windows client.

Is this work somewhere in a public repository and is there a way to contribute?

**Comment by imLinguin** [comment]
> Is this work somewhere in a public repository and is there a way to contribute?

Basic implementation is available here

https://github.com/imLinguin/comet

If you want to help reach out to us in Discord  :)

**Comment by Nocccer** [comment]
For epic games i created a new ticket whish describe how to get achivements for epic. Feel free to code this into heroic

**Comment by Arcitec** [comment]
For GOG games, achievements are supported if comet is installed/enabled.

For Epic games, you must enable "EOS Overlay" per-game to support achievements. Because Legendary (the Epic store backend) does not support achievements.

See this discussion:

https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/discussions/2728

**Comment by MrArdillo** [comment]
How is this going?

> For GOG games, achievements are supported if comet is installed/enabled. Comet isn't a complete project yet, but it seems like the goal is to integrate it into Heroic later.
> 
> For Epic games, you must enable "EOS Overlay" per-game to support achievements. Because Legendary (the Epic store backend) does not support achievements.
> 
> See this discussion:
> 
> [#2728](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/discussions/2728)

**Comment by tehmasterer** [comment]
> How is this going?

GOG achievements unlocked for me while playing Dishonored yesterday. There is no notification or sound played when in-game, but going to my GOG profile on the website shows them unlocked in the correct order. Comet support was enabled by default in the HGL settings.

**Comment by tehmasterer** [comment]
Epic achievements work too, using both the old EOS and the new "Epic Achievements" system. You can see your achievements with both systems, by pressing `Shift + F3`, provided you have the overlay enabled in HGL. 

With games that only support EOS, you will get a notification that pops out from the right of the screen when you get an achievement, though sometimes the thumbnail is garbled. With Epic Achievements, the notifications are shown at the top of the screen and a sound plays. Gameplay is also briefly slowed when the notification pops.

The only game on Epic where achievements didn't work for me was Kingdom Come: Deliverance, and I don't think that has anything to do with HGL.

**Comment by MrArdillo** [comment]
Yeah, with epic I don't have any issues right now but I was wondering if it could be possible to have something similar with gog

**Comment by ExpandingShapes** [comment]
I confirm, achievements work for me on  a GOG game too

**Comment by MrArdillo** [comment]
With the overlay and popups too?

**Comment by nicolasnobelis** [comment]
AFAIK, Heroic does not have achievement popups for Gog games till Comet adds support for it:
See https://github.com/imLinguin/comet/issues/17

Overlay and popups work fine in Heroic for Epic games.

**Comment by ExpandingShapes** [comment]
> With the overlay and popups too?

No, there were no popups and no overlay. I noticed I got achievements by going to my profile on GOG website

**Comment by lmReef** [comment]
not sure if its quite what this issue is about but overlay aside for now I opened a PR to show achievements on the game page: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/5208

just GOG for now but will look into the others soon

**Comment by Nocccer** [comment]
> not sure if its quite what this issue is about but overlay aside for now I opened a PR to show achievements on the game page: https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/pull/5208
> 
> just GOG for now but will look into the others soon

I once wrote down how to obtain achivements in epic. Don't know if this is still viable, but it is a good starting point.
#2382

**Comment by imLinguin** [comment]
In legendary fork of Rare client there is a branch we could pull to our fork to maybe offload the achievement loading logic from Heroic itself.
